memory/dma_memtest.py

```diff
@@ -0,0 +1,130 @@
+
+        if (mem / 32) < 1:
+            sim_cps += 1
+
+        return int(sim_cps)
+
+    def test(self):
+        parallel_procs = []
+        self.tmpdir = data_dir.get_tmp_dir()
+        os.chdir(self.tmpdir)
+        # This is the reference copy of the linux tarball
+        # that will be used for subsequent comparisons
+        self.log.info('Unpacking base copy')
+        base_dir = os.path.join(self.tmpdir, 'linux.orig')
+        archive.extract(self.tarball, base_dir)
+        self.log.info('Unpacking test copies')
+        for j in range(self.sim_cps):
+            tmp_dir = 'linux.%s' % j
+            if self.parallel:
+                os.mkdir(tmp_dir)
+                # Start parallel process
+                tar_cmd = 'tar jxf ' + self.tarball + ' -C ' + tmp_dir
+                self.log.info("Unpacking tarball to %s", tmp_dir)
+                obj = process.SubProcess(cmd=tar_cmd, verbose=False,
+                                         shell=True)
+                obj.start()
+                parallel_procs.append(obj)
+            else:
+                self.log.info("Unpacking tarball to %s", tmp_dir)
+                archive.extract(self.tarball, tmp_dir)
+        # Wait for the subprocess before comparison
+        if self.parallel:
+            self.log.info("Wait background processes before proceed")
+            for proc in parallel_procs:
+                proc.wait()
+        parallel_procs = []
+        self.log.info('Comparing test copies with base copy')
+        for j in range(self.sim_cps):
+            kernel_ver = os.path.basename(self.tarball).strip('.tar.bz2')
+            tmp_dir = 'linux.%s/%s' % (j, kernel_ver)
+            if self.parallel:
+                diff_cmd = 'diff -U3 -rN linux.orig/' + kernel_ver
+                diff_cmd += ' ' + tmp_dir
+                self.log.info("Comparing linux.orig with %s", tmp_dir)
+                obj = process.SubProcess(cmd=diff_cmd, verbose=False,
+                                         shell=True)
+                obj.start()
+                parallel_procs.append(obj)
+            else:
+                try:
+                    self.log.info('Comparing linux.orig with %s', tmp_dir)
+                    process.system('diff -U3 -rN linux.orig linux.%s' % j)
+                except process.CmdError, error:
+                    self.nfail += 1
+                    self.log.info('Error comparing trees: %s', error)
+
+        for proc in parallel_procs:
+            out_buf = proc.get_stdout()
+            out_buf += proc.get_stderr()
+            proc.wait()
+            if out_buf != "":
+                self.nfail += 1
+                self.log.error('Error comparing trees: %s', out_buf)
+
+        # Clean up for the next iteration
+        parallel_procs = []
+
+        self.log.info('Cleaning up')
+        for j in range(self.sim_cps):
+            tmp_dir = 'linux.%s' % j
+            shutil.rmtree(tmp_dir)
+        shutil.rmtree(base_dir)
```

Comment:
Clean up routines are expected be in `tearDown()` so you can write it only once for all tests. You have only one test function here, but I'm pointing this just to keep it sane.